File: ros_essentials/src/topic03_perception/open_copy.py
#!/usr/bin/env python

import numpy as np
import cv2

image_name = "flower"

# read an image from file
img = cv2.imread("images/"+image_name+".jpg")

# create window holder for the image
cv2.namedWindow("Image Viewer",cv2.WINDOW_NORMAL)

# display the image
cv2.imshow("Image Viewer",img)

# wait for key press before proceeding
print ('press a key inside the image to make a copy')
cv2.waitKey(0)

# write image to folder images/copy/
cv2.imwrite("images/copy/"+image_name+"-copy.jpg",img)

File: ros_essentials/src/topic03_perception/cpp/open_copy.cpp
#include <opencv2/highgui.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <stdio.h>
#include <opencv2/opencv.hpp>
#include <opencv2/core/core.hpp>

using namespace cv;

using namespace cv;
using namespace std;

 
int main( )
{
       // image data structure
       Mat image;
 
       // LOAD image from absolute path
       // Read the file "image.jpg".
       image = imread("/home/yufanana/catkin_ws/src/ros_essentials/src/topic03_perception/images/chess.jpg", cv::IMREAD_COLOR); 
              //This file "image.jpg" should be in the project folder.
              //Else provide full address : "D:/images/image.jpg"
 
       if(! image.data )  // Check for invalid input
       {
              cout <<  "Could not open or find the image" << std::endl ;
              return -1;
       }
 
       //DISPLAY image
       namedWindow( "window", WINDOW_AUTOSIZE ); // Create a window for display.
       imshow( "window", image ); // Show our image inside it.
 
       //SAVE image
       imwrite("/home/yufanana/tmp/copy_image.jpg",image);// it will store the image in name "result.jpg"
 
       waitKey(0);                       // Wait for a keystroke in the window
       return 0;
}
